third_party/amd/include/TritonAMDGPUTransforms/Passes.td

```diff
@@ -166,4 +166,21 @@ def TritonAMDGPUBlockPingpong: Pass<"tritonamdgpu-block-pingpong", "mlir::Module
   let dependentDialects = ["mlir::ROCDL::ROCDLDialect, mlir::triton::amdgpu::TritonAMDGPUDialect"];
 }
 
+def TritonAMDGPUInThreadTranspose: Pass<"tritonamdgpu-in-thread-transpose", "mlir::ModuleOp"> {
+  let summary = "Transpose K-outer dot operand while data is loaded into register right before writing to LDS";
+
+  let description = [{
+    Transpose K-outer dot operand (not consecutive on K dimension) right before writing data into LDS. This feature
```

Comment:
Avoid using "K-outer" term, which is not used anywhere. Use `M/N-continuous` or `non-K-continuous"